### ISSUE TITLE
Float repr fail in nested tuple access error message

### ISSUE THREAD

**Issue by durka** [rust-lang/rust#28243]
This code

```rust
(1, (2, 3)).1.1
```

produces this error message

```
expr.rs:5:14: 5:17 error: unexpected token: `1.1`
expr.rs:5 {(1, (2, 3)).1.1}
                       ^~~
expr.rs:5:14: 5:17 help: try parenthesizing the first index; e.g., `(foo.1).10000000000000009`
```

It should suggest `(foo.1).1` of course, but floats gotta be floats.

**Comment by christopherdumas** [comment]
How should this be fixed?

**Comment by eddyb** [comment]
I thought we kept the original string around, should be able to split that on the `.` and print the two parts without further parsing.

**Comment by christopherdumas** [comment]
I think I found where the error comes from, at libsyntax/parse/parser.rs:473. Does that seem right? If so, it doesn't look like it has access to the original string. It is constructing the error from the `as_str()` of the token.

**Comment by durka** [comment]
@christopherdumas I don't think that's where the erroneous error message is produced. It's the help text that is trying to repr the float.

**Comment by durka** [comment]
So actually line 2366 in that file.

**Comment by durka** [comment]
So it seems like insteading of parsing `fstr` and writing it back out, as @eddyb said we should be able to just split `fstr` at `.`.

**Comment by christopherdumas** [comment]
`fstr.split(".").collect::<Vec<_>>()[1..]`?

**Comment by durka** [comment]
`fstr.splitn(2, ".").last().unwrap()` seems more direct.

**Comment by christopherdumas** [comment]
Hmm. Yeah.
